lib/api/controllers/funnelController.js

```diff
@@ -710,4 +765,22 @@ function doAction(controller, request) {
   return ret;
 }
 
+/**
+ * Very straightforward function to check only if the version satisfies
+ * semver range major version
+ *
+ * @param {String} version
+ * @param {String} range
+ *
+ * @returns {Boolean}
+ */
+function satisfiesMajor (version, range) {
```

Comment:
Why not using [semver](https://www.npmjs.com/package/semver), which is already included in our package.json?

The way it's currently coded, the compatibility matrix is misleading. I, for one, would have expected a semver range, and I expect to be able to enter something like `>=7 <9` for instance, which is a probable value.

Reply:
I choose to use some kind of simplified semver range for now because semver is really slow and this method is called whenever Kuzzle receive a request.
```
Semver x 176,414 ops/sec ±4.15% (85 runs sampled)
Manual: only major x 776,075,210 ops/sec ±2.42% (90 runs sampled)
```
We won't need support for this kind of range (`>=7 <9`) until v3.

Do you think it's unnecessary ?

Reply:
Indeed, this is slow.

My real problem is the syntax similarity with semver, I find that _very_ misleading for future maintenance.

Since the matrix is internal, I think that using a custom format that has nothing to do with semver might be a better idea.

Something like:

```
{
  "js": { "min": 7 },
  "csharp": { "max": 2},
  "cpp": { "max": 2},
  "java": { "max": 3},
  "android": { "max": 5},
  "go": { "max": 3},
  "php": { "max": 4}
}
```

Of course, we would be able to have both a min and a max version. This should make the version check even faster while unambiguously being different from semver.

What do you think?